### PR TITLE
CRNLab v0.1 working

### PR DESCRIPTION
**Pull request by ayush9pandey**
The pull request addresses #30 providing CRNLab class for high level interface to internal bioCRNpyler features.

Check the jupyter notebook - Beginner User Guide to get started with and for details on CRNLab class. 

Other than the working CRNLab, this pull request also fixes some bugs all over the repository. 

Notable changes include :

* txtl.py is renamed to extracts.py and change is reflected everywhere
* `coverage run setup.py test` runs the unit test framework and reports (`coverage report`) on the coverage of the tests.
* Jupyter notebook bugs fixed and all notebooks given better names.
* SBML models are validated addressing #34 
* Jupyter notebooks cleaned and a starter guide notebook added, addressing #28 
* Minor bugs fixed at other places.

### CODE REVIEW

File: biocrnpyler/crnlab.py [hunk 1]
```diff
@@ -35,23 +36,27 @@ def mixture(self, name, **kwargs):
...
+            filename = name + str('.csv')
+            with open(filename, 'r') as f:
+                reader = csv.reader(f)
+                # Read off the parameters
+                params = {}
+                for row in reader:
+                    temp = row[1].replace('.','',1).replace('e','',1).replace('-','',1)
+                    if temp.isdigit():
+                        params[str(row[0])] = float(row[1])
+        except:
```
Comment (zoltuz) [inline biocrnpyler/crnlab.py]: It's a minor thing, but it'd better to specify what exception will be catched. Now, it catches everything.

File: .gitignore [hunk 2]
```diff
@@ -27,7 +28,6 @@ docs/_build/
 # Covers JetBrains PyCharm
 # Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
 # User-specific stuff
-.idea/**/workspace.xml
```
Comment (zoltuz) [inline .gitignore]: I'd rather keep the workspeace.xml out of the gitrepo

File: biocrnpyler/crnlab.py (hunk 1 above)
Comment (ayush9pandey) [reply biocrnpyler/crnlab.py]: I think that's the outdated file (it says so on Github, and those lines were deleted with later commits), not sure how or why they are shown now.

Comment (ayush9pandey) [reply biocrnpyler/crnlab.py]: I had parameter file reading earlier when Parameter class was not ready but then I removed all that to integrate with the Parameter class.

File: .gitignore (hunk 2 above)
Comment (ayush9pandey) [reply .gitignore]: Yes! I removed the .idea directory, probably that's why I did this. Sorry about that.

Comment (zoltuz) [reply .gitignore]: No worries, I've found another instance of this in the gitingore file. All good!

**Pull request merged**